smartsim/settings/launch_settings.py

```diff
@@ -126,6 +126,15 @@ def __init__(
             """The launcher type"""
         except ValueError:
             raise ValueError(f"Invalid launcher type: {launcher}")
+
+        if launch_args is not None:
+            if not (
+                isinstance(launch_args, t.Mapping)
+                and all(isinstance(key, str) for key, val in launch_args.items())
+            ):
+                raise TypeError(
+                    "batch_args argument was not of type mapping of str and str"
+                )
```

Comment:
Similar '`dict` vs `Mapping` inconsistency' and 'should we runtime type check the values of `launch_args`' comment here as with `BatchSettings`